grafana/grafana-infinity-datasource
# Httptrace

This is nice to have PR and allow us investigate any connection issues. More details about HTTP trace can be found in https://pkg.go.dev/net/http/httptrace

To debug connection issues such as DNS lookup, adding the HTTPTrace event debug logging.

## How to test

* `git checkout httptrace`
* `yarn && yarn build && mage -v && docker compose -f ./docker-compose-debug.yaml up`
* Open [localhost](http://localhost:3000/explore?schemaVersion=1&panes=%7B%2203z%22:%7B%22datasource%22:%22Infinity%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22yesoreyeram-infinity-datasource%22,%22uid%22:%22Infinity%22%7D,%22type%22:%22json%22,%22source%22:%22url%22,%22format%22:%22table%22,%22url%22:%22%22,%22url_options%22:%7B%22method%22:%22GET%22,%22data%22:%22%22%7D,%22root_selector%22:%22%22,%22columns%22:%5B%5D,%22filters%22:%5B%5D,%22global_query_id%22:%22%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D,%2240s%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bservice_name%3D%5C%22yesoreyeram-infinity-datasource%5C%22%7D%20%7C%3D%20%60HTTPTrace%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) which will open both a infinity query and loki query as seen in the screenshot. ( you may need to run both the queries and adjust the timerange to see the results )

<img width="1191" alt="image" src="https://github.com/user-attachments/assets/524e36ba-d06a-42f9-8d03-ab1dbb055311">

<img width="1553" alt="image" src="https://github.com/user-attachments/assets/faeac6d3-655b-425b-98a2-b8bb1c6b689c">
